p2p/node/peerManager/peerManager.go

```diff
@@ -543,12 +543,13 @@ func (pm *BasicPeerManager) calculatePeerResponsiveness(peer p2p.PeerID) float64
 }
 
 // Peers will be divided into three buckets (good, bad, ugly) based on their quality score
-func (pm *BasicPeerManager) recategorizePeer(peerID p2p.PeerID) error {
+func (pm *BasicPeerManager) recategorizePeer(peerID p2p.PeerID, topic string) error {
 	peerQuality := pm.GetTagInfo(peerID).Tags["quality"]
 	topics := pm.getPeerTopics(peerID)
+	topics[topic] = struct{}{}
 
 	// remove peer from DBs first
-	for _, topic := range topics {
+	for topic := range topics {
```

Comment:
why are you taking the index, instead of the topic?

Reply:
That is the topic